### PR TITLE
speed up annotation processing: migrate Room `kapt`->`ksp`

### PR DESCRIPTION
**Pull request by juleskers**
hi @mueller-ma ,

Thanks for the great app, I've been using it happily on my phone these last years.

I'm playing with the code to see if I can (eventually, don't get your hopes up), do something about proper multi-sim, parallel tracker support.

In the meantime, just to get my feet wet, here is an unblock for the current dependabot build failures (#271).

While doing that, my IDE told me that "ksp" is faster than kapt for annotations, so I migrated that as well.

Hope this helps!

### CODE REVIEW

**Comment by mueller-ma** [comment]
Thanks for this PR. I update Gradle in a new PR (#274). Can you rebase this PR on `main`?

**Comment by juleskers** [comment]
Ha, I was afraid that might happen.
Between the time I pulled my fork and opened the PR, I already had to rebase on top of the dependabot updates you just merged, which I *also* had in here originally 😅

I'll do the rebase, but I'm on mobile now just before bedtime, so it'll have to wait until tomorrow at the earliest.

**Comment by juleskers** [comment]
Rebase done!

Not much remains of the original dependency bumps.

I'm sorry that our workstreams crossed. I'd have loved to save you some more work :heart: 

Anyways, the annotation-speedup should still be valuable.

File: app/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 plugins {
     id "com.android.application"
     id "kotlin-android"
-    id "kotlin-kapt"
+    id 'com.google.devtools.ksp'
```
Comment (mueller-ma) [inline app/build.gradle:4]:
```suggestion
    id "com.google.devtools.ksp"
```

**Comment by mueller-ma** [comment]
FYI: I'll test Renovate Bot on a different project, because it can update gradlew automatically: https://github.com/mueller-ma/Coffee/pull/309

**Comment by juleskers** [comment]
Thanks for the quotation fix! I should really look into configuring IntelliJ for that, it's not the first time it did this to me. (If I recall correctly, it's something about string interpolation, only in double quotes)

**Pull request merged**